src/groups.jl

```diff
@@ -4,7 +4,15 @@
 
 immutable BenchmarkGroup
     tags::Vector{Any}
-    data::Dict{Any,Any}
+    data::Associative
+
+    function BenchmarkGroup(tags, data::Dict)
+        new(convert(Vector{Any}, tags), convert(Dict{Any, Any}, data))
+    end
+
+    function BenchmarkGroup(tags, data)
+        new(convert(Vector{Any}, tags), data)
+    end
 end
 
 BenchmarkGroup(tags::Vector, args::Pair...) = BenchmarkGroup(tags, Dict(args...))
```

Comment:
...and without the inner constructors, I believe we can still maintain the default behavior by changing `Dict(args...)` here to `Dict{Any,Any}(args...)`.